content/backend/typescript-apollo/9-deployment.md

```diff
@@ -0,0 +1,692 @@
+datasource db {
+  provider = "postgresql"  
+  url      = env("DATABASE_URL")  
+}
+```
+
+</Instruction>
+
+Let's understand what is happening: 
+
+1. The `provider` field signifies the underlying database type. You changed the field from `sqlite` to `postgresql`.  
+
+2. The `url` field specifies the connection string to the database. For SQLite, this is simply the file path. In the case of PostgreSQL, the database connection path will be read from the `DATABASE_URL` [environment variable](https://www.prisma.io/docs/guides/development-environment/environment-variables) of the machine's local environment. 
+
+
+Now, you need to run an instance of PostgreSQL on your local machine. You are going to do this using a containerized version of PostgreSQL. 
+
+<Instruction>
+
+Create a `docker-compose.yml` file to store your PostgreSQL container configuration:
+
+```bash(path=".../hackernews-typescript/")
+touch docker-compose.yml
+```
+</Instruction>
+
+<Instruction>
+
+```yaml{}(path=".../hackernews-typescript/docker-compose.yml")
+version: '3.8'
+services:
+
+  # Docker connection string for local machine: postgres://prisma:prisma@localhost:5432/
+
+  postgres:
+    image: postgres:10.3    # 1
+    restart: always
+    environment:            # 2
+      - POSTGRES_USER=prisma
+      - POSTGRES_PASSWORD=prisma
+    volumes:                # 3
+      - postgres:/var/lib/postgresql/data
+    ports:
+      - '5432:5432'         # 4
+
+volumes:
+  postgres:
+```
+</Instruction>
+
+Let's go through some of the import parts of this compose file:
+
+1. the `image` field represents the docker image that will be used.
+2. The `environment` array specifies which environment variables get passed to the container during initialization. This is typically used to pass configuration options and secrets (such as username and password) to the container.
+3. The `volumes` array is used for persisting data in the host file system. 
+4. The `ports` array is used to map ports from the host machine to the container. The format follows the convention of "host_port:container_port". In this case, you are mapping the port `5432` of the host machine to port `5432` of the `postgres` container. 5432 is conventionally the port used by PostgreSQL. 
+
+With that out of the way, we can spin up the database with one simple command. Before you move on to the next instruction, make sure that nothing is already running on port 5432, in which case, the following command will fail. 
+
+<Instruction>
+
+Open a *new* terminal window and run the following command:
+
+```bash(path=".../hackernews-typescript/")
+docker-compose up
+```
+
+</Instruction>
+
+If everything worked properly, the new terminal window should show logs saying that the "postgres" container has started up successfully. 
+
+> **Note:** You cannot close the new terminal window as it will also stop the container. You can avoid this if you add `-d` to the previous command, like this: `docker-compose up -d`
```

Comment:
> I would suggest using `docker-compose up -d` instead of `docker-compose up` in the Instruction block.

I considered this but opted to not use the `-d` flag because it shows all the logs of what is happening under the hood inside the container. This has educational value to new users and might be helpful in debugging errors in case something goes wrong. 

For expert users, the choice to use the `-d` flag is always there and is made explicit in the note right after the instruction block. 

I'm willing to change this though and I'd love to hear your thoughts on this!